master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductList/ProductDetail.xhtml

```diff
@@ -0,0 +1,123 @@
+<html xmlns="http://www.w3.org/1999/xhtml"
+	xmlns:f="http://xmlns.jcp.org/jsf/core"
+	xmlns:h="http://xmlns.jcp.org/jsf/html"
+	xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
+	xmlns:ic="http://ivyteam.ch/jsf/component"
+	xmlns:p="http://primefaces.org/ui"
+	xmlns:pe="http://primefaces.org/ui/extensions">
+<h:body>
+	<ui:composition template="/layouts/frame-10.xhtml">
+	<ui:define name="title">Product Detail </ui:define>
+	<ui:define name="content">
+		<script>
+			window.isHideTaskName = true;
+			window.isHideTaskAction = true;
+			window.isHideCaseInfo = true;
+		</script>
+			
+		<h:outputStylesheet name="layouts/styles/customStyle.css" />
+	<h:form id="addProductDialogForm" >
+		<div class="ui-g">
+			<div class="ui-g-4">
+				  <p:outputLabel  value="Product Detail" style="font-weight:bold;font-size:20px"/>
+			</div>
+						
+		</div>
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="productName" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/productName')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:inputText id="productName" value="#{data.product.productName}" required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/productName')}" />
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="status" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/state')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:selectOneMenu id="status" value="#{data.product.productStatus}" converter="pojoConverter" 
+									required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/state')}">
+								<f:selectItem noSelectionOption="true" />
+								<f:selectItems value="#{enumBean.productStatusValues}" var="status" itemLabel="#{status.getCmsName()}" />
+							</p:selectOneMenu>
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="validThrough" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/validThrough')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:datePicker id="validThrough" value="#{data.product.validThrough}" pattern="dd.MM.yyyy"

```

Comment:
also use CMS for the pattern="dd.MM.yyyy", I found that is use 3 times, I think we can use CMS that we can reuse pattern

Reply:
This pattern is used many times in the dialogs but it is not locale dependent, but how we want to display it all the time. My suggestion would be to leave it like it is for reasons of readability of code.

Reply:
Yes, I'd agree on that @fkalanti . It would be slightly more maintainable, as nqhoan state, but the readability will be downgraded.

I think here in the market we can accept a variety of choices as the contributor prefers. ... while we would enforce more strict rules that we agree on in larger teams.